### PR TITLE
add shutil.which backport to py3compat

### PR DESCRIPTION
**Pull request by minrk**
and use it in a few places to avoid shared dependency on utils.process

### CODE REVIEW

File: IPython/utils/py3compat.py [hunk 1]
```diff
@@ -184,6 +251,8 @@ def doctest_refactor_print(func_or_str):
     def get_closure(f):
         """Get a function's closure attribute"""
         return f.func_closure
+    
+    which = _shutil.which
```
Comment (takluyver) [inline IPython/utils/py3compat.py]: shouldn't this be `_shutil_which`?

Comment (minrk) [inline IPython/utils/py3compat.py]: ha, yes.

**Comment by takluyver** [comment]
Looks good, running the Windows tests. [3.4](https://jenkins.jupyter.org/job/windows-py3.4-prtest/6/), [2.7](https://jenkins.jupyter.org/job/windows-py2.7-prtest/9/).

**Comment by takluyver** [comment]
Myeh, test failure:

```
  File "c:\jenkins\workspace\windows-py3.4-prtest\IPython\utils\tests\test_process.py", line 55, in test_find_cmd_pythonw
    nt.assert_true(path.endswith('pythonw.exe'))
```

Spinning up my Windows VM to investigate.

**Comment by takluyver** [comment]
minrk/ipython#23 should fix it.

**Comment by takluyver** [comment]
[Good](https://jenkins.jupyter.org/job/windows-py3.4-prtest/7/) and [Good](https://jenkins.jupyter.org/job/windows-py2.7-prtest/10/) on Windows.

**Pull request merged**